src/Core/Form/IdentifiableObject/DataHandler/CategoryFormDataHandler.php

```diff
@@ -26,103 +26,29 @@
 
 namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
 
-use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Category\Command\AbstractAddCategoryCommand;
+use PrestaShop\PrestaShop\Core\Domain\Category\Command\AbstractEditCategoryCommand;
 use PrestaShop\PrestaShop\Core\Domain\Category\Command\AddCategoryCommand;
 use PrestaShop\PrestaShop\Core\Domain\Category\Command\EditCategoryCommand;
-use PrestaShop\PrestaShop\Core\Domain\Category\ValueObject\CategoryId;
-use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
-use Symfony\Component\HttpFoundation\File\UploadedFile;
+use PrestaShop\PrestaShop\Core\Domain\Category\Exception\CategoryConstraintException;
 
 /**
  * Creates/updates category from data submitted in category form
+ *
+ * @internal
  */
-final class CategoryFormDataHandler implements FormDataHandlerInterface
+final class CategoryFormDataHandler extends AbstractCategoryFormDataHandler
```

Comment:
You can remove the `final` (and internal) we try to open extendability of all our services and handlers now